mm2src/coins/utxo/utxo_common.rs

```diff
@@ -2882,6 +2899,82 @@ pub fn address_from_pubkey(
     }
 }
 
+pub async fn validate_spv_proof<T>(coin: T, tx: UtxoTx) -> Result<(), MmError<SPVError>>
+where
+    T: AsRef<UtxoCoinFields> + Send + Sync + 'static + UtxoCommonOps + MarketCoinOps,
+{
+    let client = match &coin.as_ref().rpc_client {
+        UtxoRpcClientEnum::Native(_) => return Ok(()),
+        UtxoRpcClientEnum::Electrum(electrum_client) => electrum_client,
+    };
+    if tx.outputs.is_empty() {
```

Comment:
> Maybe the block of code that gets the transaction height starting from here can be made into a function `get_tx_height`, this can be useful in other places to reduce boilerplate code like here
> 
> https://github.com/KomodoPlatform/atomicDEX-API/blob/fa9d3e661dc1afc1258602ecf8bc141bc914e9cc/mm2src/coins/lightning/ln_utils.rs#L561-L569

True but we do not have exactly the same logic, in this function you have the following condition `if item.tx_hash == rpc_txid { ` that I don't need/use - so maybe we can brainstorm a common one together.

Reply:
I think `get_tx_height` should return the height here instead of using `break; `https://github.com/KomodoPlatform/atomicDEX-API/blob/676d30c40ad8075962e9ccf07ba551c75904ce6a/mm2src/coins/utxo/utxo_common.rs#L2971-L2972
If no height is found it should return an error which will make you avoid this check https://github.com/KomodoPlatform/atomicDEX-API/blob/676d30c40ad8075962e9ccf07ba551c75904ce6a/mm2src/coins/utxo/utxo_common.rs#L2914